src/plugins/controls/public/types.ts

```diff
@@ -16,14 +16,14 @@ import {
   IEmbeddable,
 } from '@kbn/embeddable-plugin/public';
 import { DataPublicPluginStart } from '@kbn/data-plugin/public';
-import { DataView, DataViewField, DataViewsPublicPluginStart } from '@kbn/data-views-plugin/public';
+import { DataViewField, DataViewsPublicPluginStart } from '@kbn/data-views-plugin/public';
 import { UnifiedSearchPublicPluginStart } from '@kbn/unified-search-plugin/public';
 import { ControlInput } from '../common/types';
 import { ControlsService } from './services/controls';
 
 export interface CommonControlOutput {
   filters?: Filter[];
-  dataViews?: DataView[];
+  dataViewId?: string;
```

Comment:
Why the switch to `dataViewId` rather than just `dataView` directly? Just for... initial speed perhaps, since it's only needed when changes are actually made to the control's selections?

As far as why it's no longer an array - I assume that's because a control can only have a single data view? Just wanted to clarify :)

Reply:
This is actually a _little_ bit of an awkward change introduced by this PR. Basically embeddables have this pattern where they communicate the data views they use to the Dashboard so that the dashboard can pick appropriate data views when creating a filter. The problem with this is that the data view they were communicating was a whole **data view object instance** which is non-serializable and cannot be stored in Redux. For this reason I made the switch for _Controls only_ to communicate the data view Id instead.

See [this issue](https://github.com/elastic/kibana/issues/137493) for a description of how we could better extend this pattern to other embeddables.

Reply:
Ahhh that makes perfect sense - thank you! I definitely think reducing the amount of times we have to fetch the data view objects would be a huge win for performance, so communicating IDs and only fetching when **absolutely necessary** is a great pattern 💪